asm/vm/asm_vm

```diff
@@ -648,11 +795,68 @@ validate_asm_installation() {
     fatal "ASM ${ASM_RELEASE} version is not found. Please install ASM ${ASM_RELEASE} and retry."
   fi
 
-  if [[ "${EXPANSION_GATEWAY_INSTALLED}" -eq 0 ]]; then
+  if [[ "${AUTOREGISTRATION_ENABLED}" -eq 0 ]]; then
     { read -r -d '' MSG; fatal "${MSG}"; } <<EOF || true
+VM auto-registration is not enabled in any of your ASM installation in the
+cluster. Please use install_asm script to install and make sure none of the
```

Comment:
I'll defer to @maasen but it feels a bit too implementation specific to specify `install_asm` as the install method. Can we just say "Please install ASM and ensure any values in a custom overlay don't override default values for X"?

Reply:
I would prefer that as well. I think it is OK to self reference (ie `asm_vm`) but would prefer being more focused on the outcome (ensuring value is set during installation) vs reference the specific mechanism.

Reply:
reworded.